components/automate-gateway/handler/deployment.go

```diff
@@ -48,3 +48,34 @@ func (s *DeploymentServer) ServiceVersions(ctx context.Context, _ *deployment.Se
 
 	return &deployment.ServiceVersionsResponse{Services: gatewayServices}, nil
 }
+
+func (s *DeploymentServer) GetDeploymentStatus(ctx context.Context, _ *deployment.DeploymentStatusRequest) (*deployment.DeploymentStatusResponse, error) {
+	status, err := s.deploymentClient.Status(ctx, &api.StatusRequest{})
+	if err != nil {
+		logrus.WithError(err).Error("failed to get deployment status")
+		return nil, err
+	}
+
+	serviceStatus := make([]*deployment.ServiceStatus, len(status.ServiceStatus.Services))
+	oks := 0
```

Comment:
If you wanted, rather than tracking the count you could track the bool directly. Perhaps something like:

```golang
overallOk = overallOK && (svc.State.String() == "OK")
```

but it also looks correct as-is.